src/main/scala/org/hammerlab/guacamole/readsets/ReadSets.scala

```diff
@@ -0,0 +1,73 @@
+package org.hammerlab.guacamole.readsets
+
+import org.apache.spark.SparkContext
+import org.bdgenomics.adam.models.SequenceDictionary
+import org.hammerlab.guacamole._
+import org.hammerlab.guacamole.reads.Read
+
+case class ReadSets(readsRDDs: PerSample[ReadsRDD],
+                    sequenceDictionary: SequenceDictionary,
+                    contigLengths: ContigLengths) extends PerSample[ReadsRDD] {
+  override def length: Int = readsRDDs.length
+  override def apply(idx: Int): ReadsRDD = readsRDDs(idx)
+
+  val mappedReads = readsRDDs.map(_.mappedReads)
```

Comment:
I realize Spark won't run this unless it is accessed but just checking that there's definitely no performance implications to this, e.g. memory set aside? Might want to make `lazy` if we're not sure